JunList.java

```diff
@@ -19,99 +21,104 @@ public void append(int value) {
+            this.list = newList;
+        } else {
+            for (int i = trackListPosition; i > position; i--) {
+                list[i] = list[i - 1];
+            }
+            this.list[position] = value;
         }
+        trackListPosition += 1;
+    }
 
-        public void prepend(int value){
-            //creates list twice as big when necessary
-            if (trackListPosition == list.length) {
-                int[] newList = new int[list.length * 2];
-                for (int i = 0; i < list.length; i++) {
-                    if (i == 0) {
-                        newList[i] = value;
-                    } else {
-                        newList[i] = list[i - 1];
-                    }
+    public void prepend(int value) {
+        //creates list twice as big when necessary
+        if (trackListPosition == list.length) {
+            int[] newList = new int[list.length * 2];
+            for (int i = 0; i < list.length + 1; i++) {
+                if (i == 0) {
+                    newList[i] = value;
+                } else {
+                    newList[i] = list[i - 1];
                 }
-                this.list = newList;
-            } else {
-                for (int i = trackListPosition; i >= 0; i--) {
-                    if (i == 0) {
-                        list[i] = value;
-                    } else {
-                        list[i] = list[i - 1];
-                    }
+            }
+            this.list = newList;
+        } else {
+            for (int i = trackListPosition; i >= 0; i--) {
+                if (i == 0) {
+                    list[i] = value;
+                } else {
+                    list[i] = list[i - 1];
                 }
             }
-            trackListPosition += 1;
         }
+        trackListPosition += 1;
+    }
 
-        public void remove(int position){
-            for (int i = position; i < list.length - 1; i++) {
-                list[i] = list[i + 1];
-            }
-            list[list.length - 1] = 0;
-            trackListPosition -= 1;
+    public void remove(int position) {
+        for (int i = position; i < list.length - 1; i++) {
+            list[i] = list[i + 1];
         }
+        list[list.length - 1] = 0;
+        trackListPosition -= 1;
+    }
 
-        public void empty() {
-            this.list = new int[1];
-            trackListPosition = 0;
-        }
+    public void empty() {
+        this.list = new int[1];
+        trackListPosition = 0;
+    }
 
-        //return size of list
-        public int size() {
-            return trackListPosition;
-        }
+    //return size of list
+    public int size() {
+        return trackListPosition;
+    }
 
-        //check if list contains value
-        public boolean contains(int value){
-            for (int j : list) {
-                if (j == value) {
-                    return true;
-                }
+    //check if list contains value
+    public boolean contains(int value) {

```

Comment:
This logic seems to be correct but one thing downstream effect of your earlier bug is that this contains method tells me that it does contain the value 5, which logically it probably does but it shouldn't.